src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -113,6 +114,22 @@ public AuthorizeFilter(string policy)
             }
 
             var effectivePolicy = Policy;
+                            
+            var mvcOptions = context.HttpContext.RequestServices.GetRequiredService<IOptions<MvcOptions>>().Value;
+            if (mvcOptions.CombineAuthorizeFilters)
+            {
+                // Combine all authorize filters into single effective policy that's only run on the first filter
+                if (context.Filters.First(f => f is AuthorizeFilter) != this)
```

Comment:
`if (!context.IsEffectivePolicy<AuthorizeFilter>(this))`

Reply:
Yeah, use that method, You actually need to check for last instead of first.